core/src/test/scala/unit/kafka/log/LogTest.scala

```diff
@@ -199,6 +199,23 @@ class LogTest {
   }
 
   @Test
+  def testSizeForLargeLogs(): Unit = {
+    val megaByte = 1024 * 1024
+    val log = createLog(megaByte, messagesPerSegment = 1024)
```

Comment:
Yes, it would be nice if it was easier to pass the mocked log segments to `Log`, but it doesn't seem that there's an easy way to do that. In any case, this is mostly a regression test.

It would also be useful to have a test that exercises the retention by bytes code path with large logs, but we can do that in a separate PR. Would you be interested to do that?